hedera-node/test-clients/src/main/java/com/hedera/services/bdd/junit/InProcessHapiTestNode.java

```diff
@@ -170,9 +170,7 @@ public void waitForActive(long seconds) throws TimeoutException {
     @Override
```

Comment:
@tinker-michaelj Regarding https://github.com/hashgraph/hedera-services/pull/8733#discussion_r1334995163 - debugging it looks like the `th.hedera` is null in the first couple of iterations. I'm guessing that there is some time needed for it to be initialized.
Extracting the `th != null && t.hedera != null` out of the loop breaks the `waitForActive` and we are timing out when we try to start the node

Reply:
That is concerning. If `th` and probably the content of `th` is changed on a different thread, we must ensure it is thread-safe. As a minimum, `th` must be volatile. I am unsure if this is sufficient. Can the start() and shutdown() methods be called from different threads?

Reply:
I think this shouldn't happened but I'm not 100% sure. @rbair23 any thoughts?

Reply:
Currently, the in-process node runs on the HapiTest junit test runner thread (and only when debugging locally, it is never used in CI); nothing else will try to start or shut down this node.  Unless we start _debugging_ Hapi Test specs in parallel, we're pretty safe with a single-thread assumption here.